tests/Unit/Bridge/WorseReflection/Formatter/FunctionLikeSnippetFormatterTest.php

```diff
@@ -0,0 +1,140 @@
+<?php
+
+namespace Phpactor\Completion\Tests\Unit\Bridge\WorseReflection\Formatter;
+
+use ArrayIterator;
+use Phpactor\Completion\Bridge\WorseReflection\Formatter\FunctionLikeSnippetFormatter;
+use Phpactor\Completion\Core\Formatter\ObjectFormatter;
+use Phpactor\Completion\Tests\TestCase;
+use Phpactor\WorseReflection\Core\DefaultValue;
+use Phpactor\WorseReflection\Core\Name;
+use Phpactor\WorseReflection\Core\Reflection\Collection\ReflectionParameterCollection;
+use Phpactor\WorseReflection\Core\Reflection\ReflectionFunction;
+use Phpactor\WorseReflection\Core\Reflection\ReflectionFunctionLike;
+use Phpactor\WorseReflection\Core\Reflection\ReflectionMethod;
+use Phpactor\WorseReflection\Core\Reflection\ReflectionParameter;
+use Prophecy\PhpUnit\ProphecyTrait;
+
+final class FunctionLikeSnippetFormatterTest extends TestCase
+{
+    use ProphecyTrait;
+
+    /**
+     * @dataProvider provideReflectionToFormat
+     */
+    public function testFormat(ReflectionFunctionLike $reflection, $expected): void
+    {
+        $this->assertEquals(
+            $expected,
+            $this->format($reflection)
+        );
+    }
+
+    public function provideReflectionToFormat(): iterable
+    {
+        yield 'Function without parameters' => [
+            $this->reflectFunction('func', []),
```

Comment:
in general I'm using the real reflector, something like:

```
ReflectionBuilder::create()->build()->reflectFunctionsIn('<?php function foobar() {}')->first()
```

...would get you a function. Using Prophecy is OK - but "don't mock what you don't own" means we're testing what we think will happen, not what will happen (and in more complicated cases you don't want mocking hell for ReflectionClass and it's numerous related classes)

But again, no need to refactor necessarily.the mocking isn't extreme here.

Reply:
> "don't mock what you don't own" 

Didn't knew that one, I generally mock when there is dependencies involved or complex logic because I don't want to test neither of that.

I was more afraid by the reflection constructors that anything else, if you have a way to achieve it easily I'm can do it.

Reply:
```
$reflectionFunction = ReflectionBuilder::create()->build()->reflectFunctionsIn('<?php function foobar() {}')->first()
```
:)

Reply:
I can't find `ReflectionBuilder` anywhere do I miss something ?
I tried in the phpactor/phpactor too in case it was a package that I needed to add in `require-dev` for phpactor/completion but nada...

Reply:
Sorry, `ReflectorBuilder` :facepalm: